HW05.Rmd

```diff
@@ -0,0 +1,145 @@
+
+#Exported the new data frame, just in case.
+write.csv(AUC_experiment_cancer_compound,"CTRP_data_all.csv")
+```
+
+## Answers to questions
+
+### Which cancer type has the lowest AUC values to the compound "vorinostat"?
+
+Based on the figure **hematopoietic and lymphoid tissue cancer** has the lowest AUC values on average. However, the lowest AUC value correspond to **upper aerodigestive tract cancer**.
+
+```{r include=TRUE}
+#First I filtered the complete date set to include only the observations involving the compound "vorinostat"
+vorinostat_data <- AUC_experiment_cancer_compound %>% filter(cpd_name=="vorinostat")
+
+#Then, I built a boxplot to visualize the difference in AUC among cancer types:
+ggplot(vorinostat_data,mapping = aes(x = cancer_type, y= area_under_curve)) +
+  geom_jitter(col="steelblue4",size=0.8)+
+  geom_boxplot(fill="steelblue4",alpha=0.6,outlier.colour = NA)+
+  theme_bw()+
+  theme(axis.title = element_text(color="black", size=12))+
+  xlab("Cancer type") +
+  ylab("Area under curve (AUC)")+
+  theme(axis.text.x = element_text(color="black", size=8, angle=45,hjust=1))
+```
+
+### Which compound is the prostate cancer cell line 22RV1 most sensitive to? (For 22RV1, which compound has the lowest AUC value?)
+
+**leptomycin B** is the compound to which the prostate cancer cell line 22RV1 is most sensitive to.
+```{r include=TRUE}
+#First I filtered the complete date set to include only the observations involving prostate cancer cell line "22RV1"
+line22RV1_data <- AUC_experiment_cancer_compound %>% filter(ccl_name=="22RV1") %>% filter(cancer_type=="prostate")
+
+#top_n works to select only the top values for a variable in a data frame (and - is used to select the minimum value instead)
+(top_n(line22RV1_data,-1,area_under_curve))$cpd_name
+
+```
+
+### For the 10 compounds that target EGFR, which of them has (on average) the lowest AUC values in the breast cancer cell lines?
+
+The coumpound with lowest AUC values in average is **afatinib**. However, I identified 11 compounds that target EGFR, not 10.
+
+```{r include=TRUE}
+#First I filtered the complete date set to have one including only compounds that target EGFR 
+#Because there can be multiple values in the "gene_symbol_of_protein_target" (separated by ;), I used "grep" instead of "=="
+#I also filtered to only include breast cancer data.
+EGFR_data <- AUC_experiment_cancer_compound %>% filter(grepl("EGFR",gene_symbol_of_protein_target)) %>%filter(cancer_type=="breast")
+
+#Then, I built a boxplot to visualize the difference in AUC among compounds:
+ggplot(EGFR_data,mapping = aes(x = cpd_name, y= area_under_curve)) +
+  geom_jitter(col="seagreen4",size=2)+
+  geom_boxplot(fill="seagreen4",alpha=0.6,outlier.colour = NA)+
+  theme_bw()+
+  theme(axis.title = element_text(color="black", size=12))+
+  xlab("Compound name") +
+  ylab("Area under curve (AUC)")+
+  theme(axis.text.x = element_text(color="black", size=8, angle=45,hjust=1))
```

Comment:
Again, I've got to commend you on the beautiful boxplot graph. I really need to refresh/update my ggplot2 game!